functional-tests/test-bed/src/test/java/org/arquillian/smart/testing/ftest/configuration/SpecificTestClassExecutionFunctionalTest.java

```diff
@@ -0,0 +1,140 @@
+
+    @Test
+    public void should_disable_extension_and_only_execute_specified_test_when_single_test_set() throws Exception {
+        // given
+        final Project project = testBed.getProject();
+
+        project.configureSmartTesting()
+                .executionOrder(NEW)
+                .inMode(ORDERING)
+            .enable();
+
+        final Collection<TestResult> expectedTestResults = project
+            .applyAsCommits("Adds new unit test");
+
+        // when
+        final Collection<TestResult> actualTestResults = project
+            .build(module)
+                .options()
+                    .withSystemProperties("scm.range.head", "HEAD", "scm.range.tail", "HEAD~", "test", "PropertiesParserTestCase", "failIfNoTests", "false")
+                .configure()
+            .run("clean", "test");
+
+        // then
+        String capturedMavenLog = project.getMavenLog();
+        assertThat(capturedMavenLog).doesNotContain(EXPECTED_LOG_PART);
+        assertThat(actualTestResults).doesNotContainAnyElementsOf(expectedTestResults).hasSize(1);
+    }
+
+    @Test
+    public void should_execute_specified_tests_with_ordering_mode_in_order_defined_by_strategy_used() throws Exception {
+        // given
+        final Project project = testBed.getProject();
+
+        project.configureSmartTesting()
+                .executionOrder(NEW)
+                .inMode(ORDERING)
+            .enable();
+
+        final Collection<TestResult> expectedTestResults = project
+            .applyAsCommits("Adds new unit test");
+
+        // when
+        final Collection<TestResult> actualTestResults = project
+            .build(module)
+                .options()
+                    .withSystemProperties("scm.range.head", "HEAD", "scm.range.tail", "HEAD~", "test", "*Properties*", "failIfNoTests", "false")
+                .configure()
+            .run("clean", "test");
+
+        // then
+        String capturedMavenLog = project.getMavenLog();
+        assertThat(capturedMavenLog).contains(EXPECTED_LOG_PART);
+        assertThat(actualTestResults).containsSequence(expectedTestResults).hasSize(3);
+    }
+
+    @Test
+    public void should_not_execute_any_test_when_test_class_used_with_selecting_mode_does_not_fit_strategy() throws Exception {
+        // given
+        final Project project = testBed.getProject();
+
+        project.configureSmartTesting()
+                .executionOrder(NEW)
+                .inMode(SELECTING)
+            .enable();
+
+        project
+            .applyAsCommits("Adds new unit test");
+
+        // when
+        final Collection<TestResult> actualTestResults = project
+            .build(module)
+                .options()
+                    .withSystemProperties("scm.range.head", "HEAD", "scm.range.tail", "HEAD~", "test", "Properties*", "failIfNoTests", "false")
+                    .ignoreBuildFailure()
+                .configure()
+            .run("clean", "test");
```

Comment:
Default goals also work. But since the Maven documentation was using test goal with `test` property. I thought that would be preferred one.